File: ce/customerengagement/on-premises/snippets/csharp/CRMV8/auditing/cs/auditing1.cs
// <snippetauditing1>


 Console.WriteLine("Enabling auditing on the organization and account entities.");

 // Enable auditing on the organization.
 // First, get the organization's ID from the system user record.
 Guid orgId = ((WhoAmIResponse)_service.Execute(new WhoAmIRequest())).OrganizationId;

 // Next, retrieve the organization's record.
 Organization org = _service.Retrieve(Organization.EntityLogicalName, orgId,
     new ColumnSet(new string[] { "organizationid", "isauditenabled" })) as Organization;

 // Finally, enable auditing on the organization.
 bool organizationAuditingFlag = org.IsAuditEnabled.Value;
 org.IsAuditEnabled = true;
 _service.Update(org);

 // Enable auditing on account entities.
 bool accountAuditingFlag = EnableEntityAuditing(Account.EntityLogicalName, true);

// </snippetauditing1>

File: ce/customerengagement/on-premises/snippets/visualbasic/CRMV8/auditingvb/vb/auditing1.vb
' <snippetauditing1>


 Console.WriteLine("Enabling auditing on the organization and account entities.")

 ' Enable auditing on the organization.
 ' First, get the organization's ID from the system user record.
 Dim orgId As Guid = (CType(_service.Execute(New WhoAmIRequest()), WhoAmIResponse)).OrganizationId

 ' Next, retrieve the organization's record.
 Dim org As Organization = TryCast(_service.Retrieve(Organization.EntityLogicalName, orgId, New ColumnSet(New String() {"organizationid", "isauditenabled"})), Organization)

 ' Finally, enable auditing on the organization.
 Dim organizationAuditingFlag As Boolean = org.IsAuditEnabled.Value
 org.IsAuditEnabled = True
 _service.Update(org)

 ' Enable auditing on account entities.
 Dim accountAuditingFlag As Boolean = EnableEntityAuditing(Account.EntityLogicalName, True)

' </snippetauditing1>